### ISSUE TITLE
[feature request] : tune the implict graphs precision

### ISSUE THREAD

**Issue by FabriceNeyret** [jared-hughes/DesModder#53]
very often, detailed implicit or fractal graphs are ugly or unfinished because of missing detail caused by the precision limit, necessary for perf issues, but with arbitrary. 
→ would be nice to have a way to tune it. ( permanent x10 or x100 precision might not be a good idea for perfs, though ).

**Comment by jared-hughes** [comment]
I have a proof of concept for this via https://gist.github.com/jared-hughes/adac228ecf81a6a455e1181e8e8e87bf. The precision is hardcoded, so a plugin should add the accuracy as a setting (This requires re-injecting the worker though).

As far as I can tell, the number (`2^14`) controls the number of quads in the furthest depth.

**Comment by FabriceNeyret** [comment]
I'm a JS newbie ( programming it mostly like C/C++ ) : you have no other way to edit a function that to replace it's pointer ? ( you cannot get the original source ? )
wasn't the "god mode" script just applying a search/replace in the source ?

**Comment by jared-hughes** [comment]
Yes, the "god mode" script just applied a search/replace in the web worker source. Note the distinction here — it would be hard to change the UI source, but the web worker is different (kinda like code that runs in a separate thread; it runs the heavy computation behind Desmos). Since the precision only matters in the web worker, the script can completely replace the web worker (like navigating to a different web page) by specifying fresh source code.

**Comment by FabriceNeyret** [comment]
So why can't you just search-replace the same way, and thus be compatible with other scripts patching the web-worker too ? ( assuming they were apply before yours, or be as polite if coming after :-) ).

**Comment by jared-hughes** [comment]
That would be great if that was built-in. But see the following lines in the godmode code:

```js
const xhr = new XMLHttpRequest
xhr.open("GET", args[0], false)
xhr.send()
const hooked = xhr.responseText
```
This sends a request to the Desmos server to get the latest worker code because the script cannot view the web worker code directly. So if several scripts adjust the web worker in this way, each grabs the original worker code, and only the most recent one has effect.

Once graph dependencies are implemented in DesModder, then it would be smart/reasonable to have a search-replace API that allows several plugins to go in order and operate sequentially on the worker code, each applying to the results of the prior plugin as you suggest.

**Comment by FabriceNeyret** [comment]
got it. I totally missed that part in the source. 

I didn't understood before what you meant by graph ( overused term in the context of DesmosGraph :-p ) dependency. Now I get it + the problem, sure, it would be a great added value of the plugin to manage such risk of incompatibility !
BTW here it is not really the problem of a blind dependency graph per se, you still want script to act differently. To me it looks more like implementing an API between Desmos and script to access resources like the worker (either this, or directly the search-replace orders ).

**Comment by jared-hughes** [comment]
This has become much more possible in light of recent improvements to DesModder's system of module overrides.
